imx.mk

```diff
@@ -0,0 +1,198 @@
+# S_USER (Secure User - TA) are compiled
+################################################################################
+override COMPILE_NS_USER   := 64
+override COMPILE_NS_KERNEL := 64
+override COMPILE_S_USER    := 64
+override COMPILE_S_KERNEL  := 64
+
+OPTEE_OS_PLATFORM = imx-mx8mqevk
+BR2_TARGET_GENERIC_GETTY_PORT = ttymxc0
+
+include common.mk
+
+################################################################################
+# Paths to git projects and various binaries
+################################################################################
+TF_A_PATH		?= $(ROOT)/trusted-firmware-a
+
+U-BOOT_PATH		?= $(ROOT)/u-boot
+U-BOOT_BIN		?= $(U-BOOT_PATH)/u-boot.bin
+
+FIRMWARE_BIN_SHA256_SUM ?= 63ec62f5d229cbed00918c8449173933f1c9d594c59396b8dd217e94f47138b0
+FIRMWARE_BIN		?= firmware-imx-8.0.bin
+FIRMWARE_BIN		?= firmware-imx-8.0.bin
+FIRMWARE_BIN_URL	?= https://www.nxp.com/lgfiles/NMG/MAD/YOCTO/$(FIRMWARE_BIN)
+FIRMWARE_VERSION	?= firmware-imx-8.0
+FIRMWARE_PATH		?= $(ROOT)/out-firmware/$(FIRMWARE_VERSION)
+LPDDR_BIN_PATH		?= $(FIRMWARE_PATH)/firmware/ddr/synopsys
+
+MKIMAGE_PATH		?= $(ROOT)/imx-mkimage
+BOOT_IMG		?= $(ROOT)/out/boot.img
+
+################################################################################
+# Targets
+################################################################################
+all: arm-tf u-boot linux optee-os buildroot flash-image
+clean: arm-tf-clean buildroot-clean u-boot-clean optee-os-clean
+
+include toolchain.mk
+
+################################################################################
+# ARM Trusted Firmware
+################################################################################
+TF_A_EXPORTS = CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+
+#	BL32=$(OPTEE_OS_HEADER_V2_BIN) \
+#	BL32_EXTRA1=$(OPTEE_OS_PAGER_V2_BIN) \
+#	BL32_EXTRA2=$(OPTEE_OS_PAGEABLE_V2_BIN) \
+
+TF_A_FLAGS  = PLAT=imx8mq SPD=opteed DEBUG_CONSOLE=1 DEBUG=0 V=1
+TF_A_FLAGS += BL32=$(ROOT)/optee_os/out/arm/core/tee-raw.bin
+
+arm-tf: optee-os u-boot
+	$(TF_A_EXPORTS) $(MAKE) -C $(TF_A_PATH) $(TF_A_FLAGS) all fip
+
+arm-tf-clean:
+	$(TF_A_EXPORTS) $(MAKE) -C $(TF_A_PATH) $(TF_A_FLAGS) clean
+
+################################################################################
+# U-Boot
+################################################################################
+
+U-BOOT_EXPORTS = CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+
+U-BOOT_DEFCONFIG_FILES := $(U-BOOT_PATH)/configs/imx8mq_evk_defconfig \
+			  $(BUILD_PATH)/kconfigs/uboot_imx8.conf
+
+$(U-BOOT_PATH)/.config: $(U-BOOT_DEFCONFIG_FILES)
+	$(U-BOOT_EXPORTS) $(MAKE) -C $(U-BOOT_PATH) imx8mq_evk_defconfig
+	(cd $(U-BOOT_PATH) && \
+                ARCH=arm64 \
+                scripts/kconfig/merge_config.sh $(U-BOOT_DEFCONFIG_FILES))
+
+.PHONY: u-boot-defconfig
+u-boot-defconfig: $(U-BOOT_PATH)/.config
+
+# The imx8mq-evk.dtb in the symlink below needs to be copied with prefix
```

Comment:
Left-over from my Makefile? https://github.com/jbech-linaro/build/blob/imx8mqevk/Makefile#L134-L135

Reply:
I'll remove it.